### PR TITLE
Working templates and updated source JSON

### PR DESCRIPTION
**Pull request by blixhavn**
Needs a lot of tidying up before any sort of merge, but feel free to have a look and comment

### CODE REVIEW

File: dojson.tpl [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # -*- coding: utf-8 -*-
 #
 # This file is part of DoJSON
-# Copyright (C) 2015 CERN.
```
Comment (jirikuncar) [inline dojson.tpl]: 2015, 2016

File: dojson.tpl [hunk 2]
```diff
@@ -28,12 +28,50 @@ from ..model import marc21
...
+      {%- if indicator2.get('name') == 'nonfiling_characters' %}
+    indicator_map2 = {str(x): str(x) for x in range(10)}
+      {%- else %}
     indicator_map2 = {{ tojson(indicator2.get('values', {})) }}
+      {%- endif -%}
     {%- endif %}
+
+    {%- set reverse_subfields = dict() %}
+    field_map = {
+      {%- for code, subfield in field.get('subfields').iteritems() %}
```
Comment (jirikuncar) [inline dojson.tpl]: items (for PY3)

File: dojson_reverse.tpl [hunk 3]
```diff
@@ -1,7 +1,7 @@
 # -*- coding: utf-8 -*-
 #
 # This file is part of DoJSON
-# Copyright (C) 2015 CERN.
+# Copyright (C) 2016 CERN.
```
Comment (jirikuncar) [inline dojson_reverse.tpl]: 2015, 2016

**Comment by blixhavn** [comment]
I guess review and merge of this PR should be done before starting on https://github.com/inveniosoftware/dojson/pull/185

File: dojson.tpl [hunk 4]
```diff
@@ -28,12 +28,50 @@ from ..model import marc21
...
+        {%- do reverse_subfields.update({clean_name(subfield['name']): code}) %}
+      {%- endfor %}
+    }
+
+    order = utils.map_order(field_map, value)
+
+    {%- if indicator1.get('name') %}
+
+    if key[3] in indicator_map1
+      {%- if indicator1.get('name') in reverse_subfields.keys() -%}
```
Comment (jirikuncar) [inline dojson.tpl:58]: if you remove `-` from `-}}` then you don't need to do `{{ " " }}` on next line.

File: src/marc_doc_2_json.py [hunk 5]
```diff
@@ -427,6 +431,20 @@ def processHTML(self, html):
             else:
                 foundIndicators = False
 
+        # Sometimes, the HTML isn't valid. RegEx it is
+        if not foundIndicators:
+            p = re.compile(r'table.*?class=\"indicators\"(.+?)\/table', re.MULTILINE | re.DOTALL)
+            tdp = re.compile(r'(\<td\>.*?\<\/td\>)', re.MULTILINE | re.DOTALL)
+            print("\n\n\n\n\nREGEX\n\n\n\n")
```
Comment (jirikuncar) [inline src/marc_doc_2_json.py]: can you remove the print statements?

File: dojson.tpl (hunk 4 above)
Comment (blixhavn) [reply dojson.tpl:58]: Won't this also add a newline?

Comment (jirikuncar) [reply dojson.tpl:58]: Probably. So let's keep it this way ...

**Pull request merged**